### ISSUE TITLE
Cypress open maintains userAgent while cypress run does not

### ISSUE THREAD

**Issue by YutoBot** [cypress-io/cypress#9319]
### Current behavior
After performing a click, `cypress run` does not respect the 'userAgent' property that is set in my plugins/index.js file, while `cypress open` does.

After further investigating, it appears that this issue occurs after the first iteration of the beforeEach() loop. The first iteration of the loop (first test) works properly; the remaining tests do not.

Our application keys off of user agent for which layout (desktop/mobile) to give to the user. 

In OPEN mode: clicking "Movies" on the footer will take me to mobile /movies-in-theaters (ok)
<img width="1361" alt="Screen Shot 2020-11-24 at 10 07 52 AM" src="https://user-images.githubusercontent.com/55967549/100134375-fcda1c00-2e3c-11eb-8233-6fffab675d82.png">

In RUN mode: clicking "Movies" on the footer will take me to desktop /movies-in-theaters (not ok)
![footer](https://user-images.githubusercontent.com/55967549/100134866-b5a05b00-2e3d-11eb-9d4f-cde0817ddd5b.png)

### Desired behavior
In RUN mode: clicking "Movies" on the footer should take me to mobile /movies-in-theaters

### Test code to reproduce

```js
// plugins/index.js

const configOverride = {};

function setUserAgent(defaultConfig) {
    const ua = defaultConfig.env.userAgent

    switch (ua) {
        case 'desktop':
            configOverride.userAgent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) 
                Chrome/64.0.3282.167 Safari/537.36';
            configOverride.testFiles = 'desktop/**.spec.js';
            break;

        case 'mobile':
            configOverride.userAgent = 'Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/602.1.50 
                (KHTML, like Gecko) CriOS/56.0.2924.75 Mobile/14E5239e Safari/602.1';
            configOverride.testFiles = 'mobile/**.spec.js';
            break;

        default:
            configOverride.userAgent = null;
    }
}

module.exports = (on, config) => {
    setUserAgent(config)
    return Object.assign({}, config, configOverride);
---

// integration/mobile/footer.spec.js

describe('Mobile Footer', () => {
    beforeEach('open mobile home page', () => {
        cy.disableModal()
        cy.visit('https://www.fandango.com')
    })

    it('can click Spotlight to land on mobile home page', () => {
        cy.get('.nav-bar').within(() => {
            cy.contains('Spotlight')
                .click()
        })

        cy.get('#now-playing-section') // this is a mobile-only element
            .should('be.visible')
    })

    it('can click Movies to land on mobile /movies-in-theaters', () => {
        cy.get('.nav-bar').within(() => {
            cy.contains('Movies')
                .click()
        })

        cy.get('.mlp__listings') // this is a mobile-only element
            .should('be.visible')
    })
})
```

### Versions
Cypress version 5.2.0
Running on macOS Mojave, 10.14.6

**Comment by jennifer-shehane** [comment]
How are you setting the `env.userAgent` during `cypress run`? Have you verified this is being set correctly by logging `Cypress.env('userAgent')`?

I can't actually recreate this issue with the given tests. Maybe the userAgent was malformed when you pasted it?

Please note that you can not change `userAgent` within a single `cypress run` test run. You'll need to perform separate `cypress run` for each userAgent you need.

**Comment by yuto89** [comment]
Hi Jennifer, apologies for missing that piece of info.

I've forked the cypress-test-tiny repo here: https://github.com/yuto89/cypress-test-tiny

I believe you should be able to reproduce the error from this simplified version.

1st test and 2nd test are identical, to showcase the first iteration bug.

`cypress:run` should result in 1 pass (first test), 2 fails
<img width="592" alt="Screen Shot 2020-11-30 at 12 57 40 PM" src="https://user-images.githubusercontent.com/55967549/100663907-c7857080-330b-11eb-9f8c-5c9b7cdb4220.png">

`cypress:open` should result in 3 passes
<img width="451" alt="Screen Shot 2020-11-30 at 12 58 36 PM" src="https://user-images.githubusercontent.com/55967549/100663926-cc4a2480-330b-11eb-8dfa-1b571f1ac271.png">

**Comment by jennifer-shehane** [comment]
@yuto89 These tests are not running as you've shown during `cypress open`. Are you missing some env vars in the test repo? Or the viewport sizes?

There was an application error, which I had to ignore. 

<img width="557" alt="Screen Shot 2020-12-16 at 9 46 05 AM" src="https://user-images.githubusercontent.com/1271364/102300772-ecbce480-3f83-11eb-8016-9ccb0795e7e4.png">

After ignoring that, none of the tests find the .navbar or display in a mobile format.

<img width="1369" alt="Screen Shot 2020-12-16 at 9 47 46 AM" src="https://user-images.githubusercontent.com/1271364/102300743-de6ec880-3f83-11eb-83a4-1be22c8dabce.png">

**Comment by yuto89** [comment]
my mistake, this should be fine as long as you run through npm 
`npm run cypress:open` and 
`npm run cypress:run`

if we don't want it through npm:

`cypress run --env userAgent=mobile`
`cypress open --env userAgent=mobile`

**Comment by cek333** [comment]
@jennifer-shehane I hope all is well.
I'm also observing some odd behaviour related to userAgent.

We're testing our application with cypress. Our application does a check to see if the browser is chrome. Previously we were using cypress 7.7.0 and all was fine. We recently upgraded to 9.5.3 and began encountering some test failures. When running the tests (via both `cypress open` and `cypress run`), the first test will pass, but the 2nd test will fail when run against the electron browser. The 2nd test encounters our "browser not supported" page from the app. In our test setup, we set the userAgent in our `cypress.json` file to be a chrome user agent string. If we test against the chrome browser instead, all the tests pass.

I created a stripped down version of the app, but it couldn't reproduce the behaviour we were seeing with the full app.

I then tried yuto89's app which tries to fetch a couple mobile pages from fandango.com. I made a couple tweaks to get it running (https://github.com/cek333/cypress-test-tiny). It does seem to exhibit some strange behaviour, though slightly different from what yuto89 reported.
* `npm run cypress:open` results:
  * If run against the electron browser, all the tests fail. Despite setting a mobile userAgent string in `plugins\index.js`, the test gets the desktop version of the fandango site. I `cy.log()` the `window.navigator.userAgent` in the `beforeEach()` and didn't see anything unusual.
  * If run against chrome, all the tests pass
* `cypress run` results:
  * If we run against electron (`npm run cypress:run`), then the first two tests pass and the third test fails (most of the time). I've attached a video of one run: https://drive.google.com/file/d/1yjB4bpJmo838wBh-XDf7hMLL9Wo5iCfY/view?usp=sharing
  * If we run against chrome (`npm run cypress:run:chrome`), then all the tests pass.

I suppose fandango may be using more than just the userAgent string to decide which page to serve up ... but it would be interesting (and perhaps related to my issue) to find out why electron can't masquerade as a mobile client.

Thanks.
